pyo3-build-config/src/errors.rs

```diff
@@ -12,15 +12,21 @@ macro_rules! ensure {
     ($condition:expr, $($args: tt)+) => { if !($condition) { bail!($($args)+) } };
 }
 
-/// Show warning. If needed, please extend this macro to support arguments.
+/// Show warning.
 #[macro_export]
 #[doc(hidden)]
 macro_rules! warn {
-    ($msg: literal) => {
-        println!(concat!("cargo:warning=", $msg))
+    ($($args: tt)+) => {
+        $crate::write_warn!(std::io::stdout(), "cargo:warning={}", format_args!($($args)+)).unwrap()
```

Comment:
Is the additional layer of `format_args!` necessary? Couldn't they just be forwarded as they are?

Reply:
This was an honest oversight :)